app/components/form/fields/NumberField.tsx

```diff
@@ -89,6 +91,8 @@ export const NumberFieldInner = <
               })}
               aria-describedby={tooltipText ? `${id}-label-tip` : undefined}
               isDisabled={disabled}
+              maxValue={max ? Number(max) : undefined}
+              minValue={min !== undefined ? Number(min) : undefined}
```

Comment:
the `min !== undefined` construct is necessary because if `min` is `0`, `min ? __ : __` evaluates to false and `minValue` is set as `undefined`